src/os/vxworks/src/os-impl-filesys.c

```diff
@@ -295,22 +296,62 @@ int32 OS_FileSysMountVolume_Impl(const OS_object_token_t *token)
     OS_filesys_internal_record_t *local;
     int32                         status;
     int                           fd;
+    struct stat                   stat_buf;
 
     local = OS_OBJECT_TABLE_GET(OS_filesys_table, *token);
 
     /*
-     * Calling open() on the physical device path
-     * mounts the device.
+     * For FS-based mounts, these are just a map to a some other
+     * directory in the filesystem.
+     *
+     * If it does exist then make sure it is actually a directory.
+     * If it does not exist then attempt to create it.
      */
-    fd = open(local->system_mountpt, O_RDONLY, 0644);
-    if (fd < 0)
+    if (local->fstype == OS_FILESYS_TYPE_FS_BASED)
     {
-        status = OS_ERROR;
+        if (stat(local->system_mountpt, &stat_buf) == 0)
+        {
+            if (S_ISDIR(stat_buf.st_mode))
+            {
+                /* mount point exists */
+                status = OS_SUCCESS;
+            }
+            else
+            {
+                OS_DEBUG("%s is not a directory\n", local->system_mountpt);
+                status = OS_FS_ERR_PATH_INVALID;
+            }
+        }
+        else
+        {
+            if (mkdir(local->system_mountpt, 0775) == 0)
```

Comment:
VxWorks mkdir doesn't take a mode; throws a lint error. 

https://docs.windriver.com/bundle/VxWorks_Kernel_API_Reference_6.9_1/page/1190833.html#b89517da-114c-41d9-a160-575213a358c5__mkdir

Reply:
Sorry - I should have mentioned, if you want to build you'll need to also pull the fix/workaround for VxWorks 7 compatibility in #706 (mkdir was fixed to be POSIX compliant in VxWorks 7).

Reply:
Hm, this commit https://github.com/nasa/osal/commit/25cd5dfccfdb38ab879dd661d4fab42cce7f3a21 seems to have broken vxWorks when attempting to load cFE - "Warning: module 0x56d75b0 holds reference to undefined symbol OS_WaitForStateChange_Impl." when trying to build and run the branch. I'll see if I can backport the fix to OSAL 5.1.0-rc1+dev16 and see if it fixes the issue.

Reply:
Yeah there was a bunch of PRs recently, I thought #706 + this one would work in isolation but maybe not... 

At any rate, with this change I was able to test/confirm that you can do:

`OS_FileSysAddFixedMap(&fs_id, "/", "/root");`

Which succeeds, then followed by: 

`OS_TranslatePath("/root/myfile.txt", myfile_path)`

Which in turn sets `myfile_path` to `//myfile.txt` 

Note - yes it translates to a double leading slash but that is technically OK on UNIX and VxWorks seems to accept this and open it fine as well...

Reply:
Can you confirm that the case of `OS_FileSysAddFixedMap(&fs_id, "/cf", "/cf");` works as well? Otherwise I'll get around to backporting the fix and checking it sometime tomorrow. 

I think I'll make another ticket for the NFS directory case, since it's unrelated besides both affecting OS_FileSysMountVolume_Impl  and is a change in the POSIX OSAL instead. That one is a lower priority since I can just tell our developers to not run cFS off of an NFS mounted directory.

Reply:
Looks like it works for that case as well. I'll wait till we rebase w/ the latest CFE to see if the NFS issue manifests again before making another ticket.

Reply:
Thanks, and yes please submit a separate ticket for the NFS issue.  In the meantime I am looking into getting one of my test targets booted using an NFS root, and I'll check if anything looks amiss on it.